modules/common/src/main/scala/surge/core/Controllable.scala

```diff
@@ -2,19 +2,23 @@
 
 package surge.core
 
+import scala.concurrent.Future
+
+case class ControlAck(success: Boolean, details: Map[String, Any] = Map.empty, error: Option[Throwable] = None)
+
 trait Controllable {
-  def start(): Unit
-  def restart(): Unit
-  def stop(): Unit
-  def shutdown(): Unit
+  def start(): Future[ControlAck]
+  def restart(): Future[ControlAck]
+  def stop(): Future[ControlAck]
+  def shutdown(): Future[ControlAck]
 }
```

Comment:
I think this can be made short and simple ;-)
How about the following ?
```
final case object Ack
trait Controllable {
 def start(): Future[Ack]
 def restart(): Future[Ack]
 def stop(): Future[Ack]
 def shutdown(): Future[Ack]
}
```

As for `case class ControlAck(success: Boolean, details: Map[String, Any] = Map.empty, error: Option[Throwable] = None)`, I don't see the point of embedding the details map in ControlAck (you ain't gonna need it...) or the point of `error: Option[Throwable]` (mostly because Future already has a "failed"-side so start() or stop() etc. can always return a failed future (with an exception embedded in it) since Future is already a disjunction-like type...

Reply:
done